connexion/middleware/security.py

```diff
@@ -0,0 +1,225 @@
+import logging
+import pathlib
+import typing as t
+from collections import defaultdict
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+from connexion.apis.abstract import AbstractSpecAPI
+from connexion.http_facts import METHODS
+from connexion.lifecycle import MiddlewareRequest
+from connexion.middleware import AppMiddleware
+from connexion.middleware.routing import CONNEXION_CONTEXT
+from connexion.security import SecurityHandlerFactory
+
+logger = logging.getLogger("connexion.middleware.security")
+
+
+class SecurityMiddleware(AppMiddleware):
+    """Middleware to check if operation is accessible on scope."""
+
+    def __init__(self, app: ASGIApp) -> None:
+        self.app = app
+        self.apis = {}
+        self.operations = {}
+
+    def add_api(self, specification: t.Union[pathlib.Path, str, dict], **kwargs) -> None:
+        api = SecurityAPI(specification, **kwargs)
+        self.apis[api.base_path] = api
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send):
+        if scope["type"] != "http":
+            await self.app(scope, receive, send)
+            return
+
+        try:
+            connexion_context = scope['extensions'][CONNEXION_CONTEXT]
+        except KeyError:
+            raise  # TODO
```

Comment:
Still need to define a custom error to raise here because of missing routing middleware.